nvflare/private/fed/app/client/worker_process.py

```diff
@@ -32,23 +34,37 @@
 from nvflare.private.fed.client.command_agent import CommandAgent
 
 
+def check_parent_alive(parent_pid, stop_event: threading.Event):
+    while True:
+        if stop_event.is_set():
+            break
+        # TODO: this check only valid in UNIX
+        if os.getppid() != parent_pid:
+            # if parent is not alive, kill its worker process
+            os.killpg(os.getpgid(os.getpid()), 9)
+            break
+        time.sleep(1)
+
+
 def main():
-    """Worker_process start program."""
+    """Worker process start program."""
     parser = argparse.ArgumentParser()
     parser.add_argument("--workspace", "-m", type=str, help="WORKSPACE folder", required=True)
     parser.add_argument("--startup", "-w", type=str, help="startup folder", required=True)
-
     parser.add_argument(
         "--fed_client", "-s", type=str, help="an aggregation server specification json file", required=True
     )
-
+    parser.add_argument("--parent_pid", type=int, help="parent process id", required=True)
```

Comment:
Can we just get ppid on startup and save that?

Reply:
sure, thanks for the suggestion